js/snippetPreview.js

```diff
@@ -193,4 +206,88 @@ YoastSEO_SnippetPreview.prototype.renderOutput = function() {
  */
 YoastSEO_SnippetPreview.prototype.reRender = function () {
     this.init();
+};
+
+/**
+ * used to disable enter as input. Returns false to prevent enter, and preventDefault and cancelBubble to prevent
+ * other elements from capturing this event.
+ * @param event
+ */
+YoastSEO_SnippetPreview.prototype.disableEnter = function( ev ) {
+	if(ev.keyCode === 13){
+		ev.returnValue = false;
+		ev.cancelBubble = true;
+		ev.preventDefault();
+	}
+};
+
+/**
+ * checks text length of the snippetmeta and snippettitle, shortens it if it is too long.
+ * @param event
+ */
+YoastSEO_SnippetPreview.prototype.checkTextLength = function( ev ) {
+	var text = ev.currentTarget.textContent;
+	switch(ev.currentTarget.id){
+		case "snippet_meta":
+			if(text.length > YoastSEO_config.analyzerConfig.maxMeta){
+				ev.currentTarget.__unformattedText = ev.currentTarget.textContent;
+				ev.currentTarget.textContent = text.substring(0, YoastSEO_config.analyzerConfig.maxMeta);
+				ev.currentTarget.className = "desc";
+			}
+			break;
+		case "snippet_title":
+			if(text.length > 40){
+				ev.currentTarget.__unformattedText = ev.currentTarget.textContent;
+				ev.currentTarget.textContent = text.substring(0, 40);
+				ev.currentTarget.className = "title";
+			}
+			break;
+		default:
+			break;
+	}
+};
+
+/**
+ * adds and remove the tooLong class when a text is too long.
+ * @param ev
+ */
+YoastSEO_SnippetPreview.prototype.textFeedback = function( ev ) {
```

Comment:
Would you ever want to add feedback separately from `checkTextLength?`. Seems it could be just this:

```JS
YoastSEO_SnippetPreview.prototype.checkTextLength = function( ev ) {
	switch(ev.currentTarget.id){
		case "snippet_meta":
			YoastSEO_SnippetPreview.checkTextLengthFor('desc', YoastSEO_config.analyzerConfig.maxMeta, ev)
			break;
		case "snippet_title":
			YoastSEO_SnippetPreview.checkTextLengthFor('title', 40, ev)
			break;
		default:
			break;
	}
};

YoastSEO_SnippetPreview.prototype.checkTextLengthFor = function( className, maxLength, ev ) {
	var text = ev.currentTarget.textContent;
	if(text.length > maxLength){
		ev.currentTarget.__unformattedText = ev.currentTarget.textContent;
		ev.currentTarget.textContent = text.substring(0, maxLength);
		ev.currentTarget.className = className + "tooLong";
	} else {
		ev.currentTarget.className = className;
	}
};
```